src/blocklock/BlocklockFeeCollector.sol

```diff
@@ -0,0 +1,200 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8;
+
+import {ReentrancyGuard} from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+
+import {SubscriptionAPI} from "../subscription/SubscriptionAPI.sol";
+import {CallWithExactGas} from "../utils/CallWithExactGas.sol";
+
+/// @title BlocklockFeeCollector contract
+/// @notice An abstract contract for collecting fees related to blocklock functionality
+/// @dev This contract is intended to be inherited by other contracts that need to collect fees.
+/// @dev The contract includes functionality from CallWithExactGas, ReentrancyGuard, and SubscriptionAPI.
+/// @dev Inspired by Chainlink's VRFV2PlusWrapper contract at: https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/vrf/dev/VRFV2PlusWrapper.sol
+/// @notice License: MIT
+abstract contract BlocklockFeeCollector is CallWithExactGas, ReentrancyGuard, SubscriptionAPI {
```

Comment:
Would love a README or notion page on this to understand how to set the right parameters in the right circumstances. I'm assuming we'll be interacting with this quite a lot to e.g. make usage free for some users, change fees on different chains, etc

Reply:
I'll add some documentation on this. There isn't logic for making usage free for some users but it can be done ASAP. @CluEleSsUK